### PR TITLE
fix: improvements in tests, build, dependencies, docs and network options

### PR DESCRIPTION
**Pull request by amunra**
General improvements:
* Enabled `SO_KEEPALIVE` in TCP connections.
* Updated dependencies to the latest versions. This also picked up new TLS certs from `webpki-roots`.
* Formatted code `cargo fmt`.
* Fixed linter issues observed by `cargo clippy`.
* Bumped system tests to use the latest QuestDB version [7.3.2](https://github.com/questdb/questdb)
* Relegated code generation for tests in https://github.com/questdb/questdb-client-test.git (`questdb-rs/src/tests/interop`) behind a new (disabled by default) `json_tests` feature flag.
* Fixed a new Windows build linker issue occurring with Visual Studio 2019 and 2022.
* In Rust you can now pass `chrono::DateTime` objects directly to `.column_ts` and `.at` methods. For this enable the `chrono_timestamp` feature.
* In C and C++ disabled building tests and examples by default. This works around some issues users have encountered with the `EXCLUDE_FROM_ALL` CMake setting: https://github.com/questdb/c-questdb-client/pull/39
* Upgraded to the newest version of corrosion [corrosion](https://github.com/corrosion-rs/corrosion) CMake library.
* Documentation updates.

Closes:
* https://github.com/questdb/c-questdb-client/issues/41
* https://github.com/questdb/c-questdb-client/issues/40
* https://github.com/questdb/c-questdb-client/issues/38

### CODE REVIEW

**Review by eugenels — COMMENTED** [review]
Overall, it looks good.

I wanted to suggest that you consider removing all this timestamp conversion shenanigans from the API. However, I noticed that a similar approach is used for other arguments such as TableName and ColumnName - validating arguments in an API call. What worries me about this approach is that it increases the error surface.
 ```
 /// The table name or column name contains invalid characters.
    InvalidName,

    /// The passed timestamp is invalid.
    InvalidTimestamp,
```

How I can choose timestamp resolution for:
```
///     .column_ts("e", SystemTime::now())?  // always micros ?
///     .at(SystemTime::now())?; // always nanos ?
```
?

File: questdb-rs/src/ingress/timestamp.rs [hunk 1]
```diff
@@ -1,31 +1,58 @@
 use crate::error;
 use std::time::{Duration, SystemTime, UNIX_EPOCH};
 
-const TIMESTAMP_MAX: i64 = i64::MAX - 1;
+const TIMESTAMP_NEG_BOUND: i128 = -9223372036854775808i128; // (i64::MIN * -1i64) as u128;
```
[the remark this answers is not in the document]
Comment (amunra) [reply questdb-rs/src/ingress/timestamp.rs:4]: I'll see if I can clean it up with a macro maybe. I couldn't do the i64 -> i128 cast as a constexpr.

File: .gitignore [hunk 2]
```diff
@@ -1,7 +1,9 @@
 target
 build
+cmake-build-debug
```
[the remark this answers is not in the document]
Comment (amunra) [reply .gitignore:3]: I'll add it. This is CLion compatibility.

File: questdb-rs/src/ingress/timestamp.rs [hunk 3]
```diff
@@ -1,31 +1,58 @@
...
+        -(extract_fn(UNIX_EPOCH.duration_since(time).expect("time < UNIX_EPOCH")) as i128)
+    }
+}
+
+fn sys_time_convert(
+    time: SystemTime,
+    extract_fn: impl FnOnce(Duration) -> u128,
+) -> crate::Result<i64> {
+    let number = sys_time_to_duration(time, extract_fn);
+    if (TIMESTAMP_NEG_BOUND..=TIMESTAMP_POS_BOUND).contains(&number) {
```
[the remark this answers is not in the document]
Comment (amunra) [reply questdb-rs/src/ingress/timestamp.rs:22]: ..and this sorts out your other comment as well. Thanks for the suggestion!

File: questdb-rs/src/ingress/timestamp.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (amunra) [reply questdb-rs/src/ingress/timestamp.rs:4]: These constants are now removed.

**Comment by amunra** [comment]
I've simplified the timestamps APIs: They are now explicit in both Rust and C++.

**Review by eugenels — APPROVED** [review]

**Pull request merged**